contracts/OffsetHelper.sol

```diff
@@ -488,11 +488,10 @@ contract OffsetHelper is OffsetHelperStorage {
                 balances[msg.sender][_tco2s[i]] >= _amounts[i],
                 "Insufficient TCO2 balance"
             );
-
-            balances[msg.sender][_tco2s[i]] -= _amounts[i];
-
-            IToucanCarbonOffsets(_tco2s[i]).retire(_amounts[i]);
-
+            if (_amounts[i] > 0) {
```

Comment:
We have a fix for this deployed in Mumbai for both BCT and NCT, you don't need this change anymore. Btw @danceratopz  it would be great if you could confirm that the issue is fixed for you now!

Reply:
Hi @kargakis,  it's not possible to verify whether the issue has been fixed on Polygon Mainnet with the current chain state as there are plenty of the lowest scored TCO2, TCO2-VCS-1052-2012, in the NCT Pool, https://polygonscan.com/token/0x463de2a5c6e8bb0c87f4aa80a02689e6680f72c7#balances.

As the issue only reveals itself when the lowest scored TCO2 is not present in the pool, it's probably not possible to test this issue on Mumbai either, but I didn't check (I was a bit lazy as the ABI for NCT on Mumbai has not been published https://mumbai.polygonscan.com/address/0x7beCBA11618Ca63Ead5605DE235f6dD3b25c530E#readContract).

Btw, I don't think there are liquidity pools for NCT or BCT setup on Sushiswap on Mumbai, so we never extensively used or tested the OffsetHelper there, cf https://discord.com/channels/706114892666765375/929319519430799360/1014126215877103647

Reply:
@danceratopz good point on liquidity pool setup in Mumbai, @lazaralex98 this is something we need to have in place in order for the offsetter contract to be prod-ready, do we track TODOs for that anywhere atm? 

The first TCO2 in the score array in BCT is empty, we can test this issue by redeeming BCT: https://mumbai.polygonscan.com/address/0xf2438A14f668b1bbA53408346288f3d7C71c10a1#readProxyContract
The offsetter contract btw should expose all the functions needed to offset without the need to swap so the lack of LPs in Mumbai should not block testing of this issue.

Reply:
@kargakis Sorry, Michalis, I didn't check BCT on Mumbai. Yes, I can confirm the fix, the Mumbai OffsetHelper now takes the second TCO2 in the list of scored TCO2s:
https://mumbai.polygonscan.com/tx/0xfbe23052dabc36e070d2b2b313ac9883159d3c6f2616c9ba21d0df91f6986cfb